apps/rush-lib/src/cli/logic/pnpm/PnpmShrinkwrapFile.ts

```diff
@@ -147,6 +162,28 @@ export class PnpmShrinkwrapFile extends BaseShrinkwrapFile {
     }
 
     if (!packageDescription.dependencies.hasOwnProperty(dependencyName)) {
+      if (checkOtherProjects) {
+        // this means the current temp project doesn't provide this dependency,
+        // however, we may be able to use a different version. we prefer the latest version
+        const minimumVersion: string = '0.0.0';
+        let latestVersion: string = minimumVersion;
+
+        this.getTempProjectNames().forEach((otherTempProject: string) => {
+          const otherVersion: string | undefined = this.getDependencyVersion(dependencyName, otherTempProject, false);
+          if (otherVersion) {
+            if (semver.gt(otherVersion, latestVersion)) {
```

Comment:
>gt [](start = 23, length = 2)

what will this do with uncomparable versions, e.g. prerelease tags or "git://"

Reply:
fixed

---
In reply to: [170407115](https://github.com/Microsoft/web-build-tools/pull/543#discussion_r170407115) [](ancestors = 170407115)